scr/hw6/ArrayValueCalculator.java

```diff
@@ -0,0 +1,35 @@
+package hw6;
+
+import static java.lang.Integer.parseInt;
+
+public class ArrayValueCalculator {
+    public static int doCalc(String[][] arr) {
+        int summ = 0;
+        try {
+            if (arr.length != 4) // lines
+            {
+                throw new ArraySizeException("Size !=4");
+            }
+            for (int i = 0; i < arr.length; i++) {
+                for (int j = 0; j < arr[i].length; j++) {
+                    if (arr[i].length != 4) //columns
+                    {
+                        throw new ArraySizeException("Size error");
+                    }
+                    try {
+                        summ += parseInt(arr[i][j], 10);
+                    } catch (NumberFormatException a) {
```

Comment:
In your solution possible to throw ArrayDataException for an invalid array size (ArraySizeException should be thrown because the array size is invalid, but we check before cells format). Example
String[][] arr = {
{"1", "XXX", "567", "1234"},
{"45"},
{"45", "998", "567", "1234"},
{"45", "998", "567", "456"}
};

Reply:
I tried to do something similar but it didn't work